src/NBitcoin/ProvenBlockHeader.cs

```diff
@@ -31,6 +31,9 @@ namespace NBitcoin
     /// </remarks>
     public class ProvenBlockHeader : PosBlockHeader
     {
+        /// <inheritdoc />
+        public override int CurrentVersion => 8;
```

Comment:
why this?

Reply:
That's my idea. 
So other stratis-only components can check if the header is PH without typecast.

Reply:
@fenix2222 had a call with Apro. I't a bad idea indeed because version goes to block's hash and we can't easily change it. Sorry, my bad. 
Could you please remove it?

Reply:
Sure, I can remove it, but what are we checking with version rule then? Should we just use `StratisHeaderVersionRule`?

Reply:
>Sure, I can remove it, but what are we checking with version rule then? Should we just use StratisHeaderVersionRule?

Yeah